client/src/components/ForgotPassword/CheckEmail/CheckEmailContainer.tsx

```diff
@@ -1,49 +1,33 @@
 import React from "react";
+import CheckEmail from "./CheckEmail";
+import { reduxForm, InjectedFormProps } from "redux-form";
 import { connect } from "react-redux";
-import CheckEmail from './CheckEmail';
-import {
-    setEmail,
-    checkEmail
-} from "../../../store/checkEmail/action";
+import { checkEmail } from "../../../store/checkEmail/actions";
+import validate from '../../../utils/validation'
 
-interface FormProps {
-    email: string;
-    setEmail: Function;
-    checkEmail: Function;
-}
-
-interface CheckEmailState {
-    email: string;
-}
+class CheckEmailContainer extends React.Component<any> {
+  onSubmit = (email: string) => {
+    return this.props.checkEmail(email);
+  };
 
-interface FormState {
-    checkEmail: CheckEmailState;
+  render() {
+    const { handleSubmit, submitting } = this.props;
+    return (
+      <CheckEmail
```

Comment:
Do we need to move it to separate compoment? DO we use it again some where?  Or will we? Maybe it is better to combine?

Reply:
You mean CheckEmail? We have CkeckEmailContainer, which deals with Redux, and CheckEmail, which is simply presentational component.

Reply:
yep, if this component will not be reusable, that you can move it here